### PR TITLE
Add sharing to Instagram Stories

### PR DESCRIPTION
**Pull request by MiSikora**
## Description

This PR adds sharing assets to Instagram Stories. I renamed `VideoBackgroundController` to `BackgroundAssetController` and hid it behind an interface for easier Compose UI previews.

Vertical card is a special case because it doesn't share card directly but embeds it more nicely in a Story. Horizontal and square cards do not show PC logo ATM. I asked about it here: p1722851984895389-slack-C05RR9P9RAT

## Testing Instructions

1. Share a podcast and select Instagram. Try sharing it with different cards.
2. Share an episode and select Instagram. Try sharing it with different cards.
3. Share an episode timestamp and select Instagram. Try sharing it with different cards.
4. Test sharing with a device in the landscape orientation.

## Screenshots or Screencast 

| Vertical | Horizontal | Square |
| - | - | - |
| vertical | horizontal | square |

## Checklist
- [x] ~If this is a user-facing change, I have added an entry in CHANGELOG.md~
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] ~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-android/milestone/61">7.70</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/BackgroundAssetController.kt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import kotlinx.coroutines.withContext
+
+internal interface BackgroundAssetController {
+    fun captureController(type: CardType): CaptureController
+
+    suspend fun capture(type: CardType): Result<File>
+
+    companion object {
+        private const val BITMAP_WIDTH = 1080
+        private const val BITMAP_HEIGHT = 1920
```
Comment (geekygecko) [inline modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/BackgroundAssetController.kt:31]: The code seems to be scaling the height of the image for the horizontal card but it's coming through in Instagram at the same height. Is that the normal behaviour on Instagram?



Can I confirm this isn't the final design? The colors and text sizes don't seem quite right.

**Review by geekygecko — APPROVED** [review]
Great work getting those into Instagram!

In landscape I couldn't change the format, is that expected?

Screenshot_20240806_172731_Pocket Debug

File: modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/BackgroundAssetController.kt (hunk 1 above)
Comment (MiSikora) [reply modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/BackgroundAssetController.kt:31]: > The code seems to be scaling the height of the image for the horizontal card but it's coming through in Instagram at the same height. Is that the normal behaviour on Instagram?

I'm not sure what do you mean. How would you expect horizontal card too look like? It needs to be scaled uniformly.

> Can I confirm this isn't the final design? The colors and text sizes don't seem quite right.

Colors are still being designed: pdeCcb-50w-p2#comment-4887

Font size's are correct. From top to bottom they are `H70`, `H40`, and `H70`. HR2vcQrWcS0scsolKZBitg-fi-2336_11067

**Comment by MiSikora** [comment]
> In landscape I couldn't change the format, is that expected?

Yes.

**Pull request merged**